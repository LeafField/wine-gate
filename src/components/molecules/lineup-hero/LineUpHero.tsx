import React, { FC } from "react";
import HeroHeader from "../../atoms/hero-header/HeroHeader";
import Image from "next/image";
import lineUpImage from "../../../images/lineup_image.jpg";

const LineUpHero: FC = () => {
  return (
    <div className="tablet:h-75 relative h-40">
      <HeroHeader title="ワイン検索" />
      <div className="clip absolute inset-0 h-full w-full">
        <div className="tablet:h-75 fixed inset-0 h-40 w-full">
          <Image
            src={lineUpImage}
            alt="様々な種類のグラスワインが並んでいる"
            fill
            style={{
              objectFit: "cover",
            }}
            sizes="100vw"
          />
        </div>
      </div>
    </div>
  );
};

export default LineUpHero;
