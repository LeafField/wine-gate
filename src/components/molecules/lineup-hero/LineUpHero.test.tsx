import { screen, render } from "@testing-library/react";
import LineUpHero from "./LineUpHero";

jest.mock("next/navigation");

describe("LineUpHeroの単体テスト", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  afterAll(() => {
    jest.restoreAllMocks();
  });

  test("LineUpHeroコンポーネントが存在することを確認する", () => {
    const { container } = render(<LineUpHero title="ワイン検索" />);
    expect(screen.getByText("ワイン検索")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
