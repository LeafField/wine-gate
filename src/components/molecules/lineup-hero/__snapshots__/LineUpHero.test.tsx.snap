// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineUpHeroの単体テスト LineUpHeroコンポーネントが存在することを確認する 1`] = `
<div>
  <div
    class="relative h-40 hero tablet:h-75"
  >
    <div
      class="absolute inset-0 z-front flex h-full w-full items-center justify-end pr-6 hero-mobile-verlay tablet:hero-overlay"
    >
      <h2
        class="text-heading2 font-bold text-white"
      >
        ワイン検索
      </h2>
    </div>
    <div
      class="absolute inset-0 h-full w-full clip"
    >
      <div
        class="absolute inset-0 h-40 w-full tablet:h-75"
      >
        <img
          alt="様々な種類のグラスワインが並んでいる"
          data-nimg="fill"
          decoding="async"
          fetchpriority="high"
          sizes="100vw"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
        />
      </div>
    </div>
  </div>
</div>
`;
